numba/np/arraymath.py

```diff
@@ -489,6 +491,7 @@ def return_false(a):
 
 
 @overload(np.min)
+@overload(np.amin)
```

Comment:
Two items:
1. In NumPy 1.25 `amin` and `min` cease to be aliases, as `amin` is actually declared as supported by Numba in the docs, the unit tests for `min` should also be applied to `amin` (same applies to the `amax`/`max` aliasing).
2. For the rest of the `@overload` additions in this module which have their aliases added, I don't think these are aliases by `id` i.e. the same function under a different name, they are just "equivalent functions", i.e. do the same thing, so they wouldn't have been supported by Numba as aliases. Essentially, for the function X and its potential alias Y, if  `X is Y`, returns `True` using NumPy 1.24 then it's a "true" alias and Numba needs support patching in for 1.25, if it's `False` then it does not.

Reply:
1. Any particular tests?
2. What is the action you want me to take?

Reply:
1. Suggest finding important cases involving `np.min` and `np.max`. Perhaps look in `test_array_reductions.py` as that's likely where they are used. It doesn't need to be exhaustive, just something to make sure that if anything "happens" to these currently untested aliases (`np.amax` and `np.amin`) it's picked up.
2. Suggest running the `X is Y` test for all the deprecated aliased functions and make sure that anything that is a true alias (function at same address) has an `@overload` declared in this patch, anything which is not a true alias needs the `@overload` removing from this patch.

Reply:
1. Added
2. `X is Y` is `False` for all added overloads in the current patch.

Reply:
I've done what I hope is an exhaustive search for 2. The summary is as follows, if you observe the same I suggest updating this patch to reflect this.

The word `alias` herein means the test `X is Y` returns `True`.

From https://github.com/numpy/numpy/pull/23314:
* `product` - deprecated in 1.25, does not alias `prod` in 1.24 or 1.25.
* `cumproduct` - deprecated in 1.25, does not alias `cumprod` in 1.24 or 1.25.
* `sometrue` - deprecated in 1.25, does not alias `any` in 1.24 or 1.25.
* `alltrue` - deprecated in 1.25, does not alias `all` in 1.24 or 1.25.

From https://github.com/numpy/numpy/pull/23302:
* `round_` - deprecated in 1.25, aliases `round` in 1.24, does not alias `round` in 1.25.
* `around` - NOT deprecated in 1.25, does not alias `round` in 1.24 or 1.25.
* `amin` - NOT deprecated in 1.25, aliases `min` in 1.24, does not alias `min` in 1.25.
* `amax` - NOT deprecated in 1.25, aliases `max` in 1.24, does not alias `max` in 1.25.

Summary of anticipated impact on this PR/NumPy 1.25 support:
1. Numba does not need to add `@overload`s in the 1.25 support for `product`, `cumproduct`, `sometrue`, or `alltrue`, these are:
   a. not aliased functions as `X is Y` is `False` in 1.24 and 1.25.
   b. not declared as supported in Numba's documentation.
2. Numba does not need to add an `@overload` in the 1.25 support for `around` as it is:
   a. not an aliased function as `around is round` is `False` in 1.24 and 1.25.
   b. an `@overload` is already present in the code base and it's declared as supported in Numba's documentation.
3. Numba does need an `@overload` *and tests* adding in the 1.25 support for `round_` even thought it is deprecated as:
   a. this is an alias in 1.24 as `round_ is round` is `True`.
   b. this is not an alias in 1.25 as `round_ is round` is `False`.
   c. Numba declares `round_` as supported in its documentation.
4. Numba does need an `@overload` *and tests* adding in the 1.25 support for `amax` and `amin` as:
   a. these are aliased functions in 1.24 as `X is Y` is `True`.
   b. these are not aliased functions in 1.25 as `X is Y` is `False`.
   c. Numba declares `amax` and `amin` as supported in its documentation.

Reply:
Thanks for the detail summary. It make sense.

Reply:
I believe I've corrected this as requested.

Reply:
Thank you, I've taken a look and as of 0c086cf the diff looks like it covers the above details.